extensions/wikia/Wall/WallMessage.class.php

```diff
@@ -1679,13 +1679,12 @@ public function canMove( User $user, $shouldLogBlockInStats = true ) {
 	}
 
 	/**
-	 * @desc Creates wall message title (a board, a thread, a message) instance and calls purgeSquid() on it
+	 * @desc calls purgeSquid() on $title instance
 	 * The flow then goes to TitleGetSquidURLs hook which cleans the list of URLs in Wall and Forum
 	 */
 	public function purgeSquid() {
-		$title = Title::newFromID( $this->getId() );
-		if ( $title instanceof Title ) {
-			$title->purgeSquid();
+		if ( $this->title instanceof Title ) {
+			$this->title->purgeSquid();
```

Comment:
`WallMessage::purgeSquid` is called when thread state is changed / message post is deleted, restored or undo is performed.

In all such cases invalidation of CDN cache and bump of `ETag` value (via `Title::invalidateCache`) is required. What if we move these two calls here?

```php
$mw->getWall()->getTitle()->invalidateCache();
wfWaitForSlaves();
```